tests/parser.cpp

```diff
@@ -2131,17 +2422,54 @@ uprobe:asdf:Stream {} tracepoint:only_one_arg {}
 
 TEST(Parser, config_error)
 {
-  test_parse_failure("i:s:1 { exit(); } config = { BPFTRACE_STACK_MODE=perf }");
-  test_parse_failure("config = { exit(); } i:s:1 { exit(); }");
-  test_parse_failure("config = { @start = nsecs; } i:s:1 { exit(); }");
+  test_parse_failure("i:s:1 { exit(); } config = { BPFTRACE_STACK_MODE=perf }",
+                     R"(
+stdin:1:19-25: ERROR: syntax error, unexpected config, expecting {
+i:s:1 { exit(); } config = { BPFTRACE_STACK_MODE=perf }
+                  ~~~~~~
+)");
+  test_parse_failure("config = { exit(); } i:s:1 { exit(); }", R"(
+stdin:1:12-16: ERROR: syntax error, unexpected call, expecting identifier
+config = { exit(); } i:s:1 { exit(); }
+           ~~~~
+)");
+  test_parse_failure("config = { @start = nsecs; } i:s:1 { exit(); }", R"(
+stdin:1:12-18: ERROR: syntax error, unexpected map, expecting identifier
+config = { @start = nsecs; } i:s:1 { exit(); }
+           ~~~~~~
+)");
   test_parse_failure("BEGIN { @start = nsecs; } config = { "
-                     "BPFTRACE_STACK_MODE=perf } i:s:1 { exit(); }");
+                     "BPFTRACE_STACK_MODE=perf } i:s:1 { exit(); }",
+                     R"(
+stdin:1:27-33: ERROR: syntax error, unexpected config, expecting {
+BEGIN { @start = nsecs; } config = { BPFTRACE_STACK_MODE=perf } i:s:1 { exit(); }
+                          ~~~~~~
+)");
   test_parse_failure("config = { BPFTRACE_STACK_MODE=perf "
-                     "BPFTRACE_MAX_PROBES=2 } i:s:1 { exit(); }");
+                     "BPFTRACE_MAX_PROBES=2 } i:s:1 { exit(); }",
+                     R"(
+stdin:1:37-56: ERROR: syntax error, unexpected identifier, expecting }
+config = { BPFTRACE_STACK_MODE=perf BPFTRACE_MAX_PROBES=2 } i:s:1 { exit(); }
+                                    ~~~~~~~~~~~~~~~~~~~
+)");
   test_parse_failure("config = { BPFTRACE_STACK_MODE=perf } i:s:1 { "
-                     "BPFTRACE_MAX_PROBES=2; exit(); }");
-  test_parse_failure("config { BPFTRACE_STACK_MODE=perf } i:s:1 { exit(); }");
-  test_parse_failure("BPFTRACE_STACK_MODE=perf; i:s:1 { exit(); }");
+                     "BPFTRACE_MAX_PROBES=2; exit(); }",
+                     R"(
+stdin:1:47-67: ERROR: syntax error, unexpected =, expecting }
+config = { BPFTRACE_STACK_MODE=perf } i:s:1 { BPFTRACE_MAX_PROBES=2; exit(); }
+                                              ~~~~~~~~~~~~~~~~~~~~
+)");
+  test_parse_failure("config { BPFTRACE_STACK_MODE=perf } i:s:1 { exit(); }",
+                     R"(
+stdin:1:8-9: ERROR: syntax error, unexpected {, expecting =
+config { BPFTRACE_STACK_MODE=perf } i:s:1 { exit(); }
+       ~
+)");
+  test_parse_failure("BPFTRACE_STACK_MODE=perf; i:s:1 { exit(); }", R"(
```

Comment:
nit: I know the rest of the file doesn't adhere to this but I like a little breathing room (line break) between the `test_parse_failure`s 🌬️

Reply:
Good point, definitely easier to read this way